core/src/integTest/scala/water/sparkling/itest/local/DataFrameConverterBenchSuite.scala

```diff
@@ -0,0 +1,120 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package water.sparkling.itest.local
+
+import org.apache.spark.SparkContext
+import org.apache.spark.h2o.testdata.{DenseVectorHolder, SparseVectorHolder}
+import org.apache.spark.h2o.utils.BenchUtils.bench
+import org.apache.spark.h2o.utils.{SharedH2OTestContext, TestFrameUtils}
+import org.apache.spark.ml.linalg.{DenseVector, SparseVector}
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+import water.api.TestUtils
+import water.sparkling.itest.local.schemas._
+
+@RunWith(classOf[JUnitRunner])
+class DataFrameConverterBenchSuite extends FunSuite with SharedH2OTestContext {
+  override def createSparkContext = new SparkContext("local-cluster[2, 1, 2048]", this.getClass.getSimpleName, defaultSparkConf)
+
+  val settings = TestFrameUtils.GenerateDataFrameSettings(
+    numberOfRows = 8000,
+    rowsPerPartition = 500,
+    maxCollectionSize = 100,
+    nullProbability = 0.0
+  )
+
+  test("Measure performance of conversion to H2OFrame on a flat data frame") {
+    testPerSchema(FlatSchema)
+  }
+
+  test("Measure performance of conversion to H2OFrame on a data frame with nested structs") {
+    testPerSchema(StructsOnlySchema)
+  }
+
+  test("Measure performance of conversion to H2OFrame on a data frame with flat arrays") {
+    testPerSchema(FlatArraysOnlySchema)
+  }
+
+  def testPerSchema(schemaHolder: TestFrameUtils.SchemaHolder): Unit = {
+    val df= TestFrameUtils.generateDataFrame(spark, schemaHolder, settings)
+    val result = bench(iterations = 5, warmup = 1) {
+      val hf = hc.asH2OFrame(df)
+      hf.remove()
+    }
+    println(result.show())
```

Comment:
I think we should move this from integTests to a new collection directory called `core/src/bench/scala/water/sparkling/bench/DataFrameConverterBenchSuite.scala` and create task on each scala gradle project `./gradlew bench` which would execute all benchmarks in each suproject. 

Also we should ideally report the outcome of bechmarks in a better way -> create some report which would be also archived on jenkins.

There are also libraries for Gralde which can help with setting up microbenchmarking infra - for example https://github.com/melix/jmh-gradle-plugin

Reply:
WDYT about using the new namespace `ai.h2o.sparkling` for the microbenchmarks?

Reply:
perfect!